protobuf/src/main/java/io/grpc/protobuf/ProtoUtils.java

```diff
@@ -107,7 +107,8 @@ private T parseFrom(InputStream stream) throws InvalidProtocolBufferException {
    */
   public static <T extends Message> Metadata.Key<T> keyForProto(final T instance) {
     return Metadata.Key.of(
-        instance.getDescriptorForType().getFullName() + Metadata.BINARY_HEADER_SUFFIX,
+        instance.getDescriptorForType().getFullName().replace('.', '-')
```

Comment:
Or.... unless they are being used in auto generated code.   :/  This will probably break someone.